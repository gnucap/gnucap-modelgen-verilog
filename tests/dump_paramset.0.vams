
`include "sub/disciplines.few"
`pragma modelgen noexpand-paramset

(* attribute="proto" *)
module resistor(a,b)
	electrical a, b;

	(*attrib=yes*) parameter real r = 0.;
	parameter real a = 0, b = 0.;

	analog begin
		I(a,b) <+ V(a,b)/r;
	end
endmodule

(* attribute="stub" *)
paramset const_res resistor;
	parameter real rr = 0., RR = 0.;
	localparam real ll = rr;
	(*attrib=yes*) parameter real xx = 0.;
	.r = 1;
	.a = xx + ll;
endparamset;


