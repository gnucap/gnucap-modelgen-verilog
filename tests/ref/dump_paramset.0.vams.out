nature current
  access = "I";
  units = "A";
  abstol = "0.001234";
endnature

nature voltage
  access = "V";
  units = "V";
  abstol = "0.0004321";
endnature


discipline electrical
  potential "voltage";
  flow "current";
enddiscipline


(* attribute="proto" *)
module resistor(a, b);
  electrical a, b;
  parameter real r = 1;

  analog begin
    I(a, b) <+ (V(a, b) / r);
  end

endmodule



(* attribute="stub" *)
paramset const_res resistor;
endparamset;

